File: base/src/vsdk/toolkit/media/ShapeDescriptor.java
//===========================================================================
//=-------------------------------------------------------------------------=
//= Module history:                                                         =
//= - May 18 2007 - Oscar Chavarro: Original base version                   =
//===========================================================================

package vsdk.toolkit.media;

public abstract class ShapeDescriptor extends MediaEntity
{
    protected String label;

    public ShapeDescriptor(String label)
    {
        setLabel(label);
    }

    public String getLabel()
    {
        return label;
    }

    public void setLabel(String label)
    {
        this.label = new String(label);
    }

    public double [] getFeatureVector() {
        return null;
    }

    public void setFeatureVector(double vector[]) {
        ;
    }
}

//===========================================================================
//= EOF                                                                     =
//===========================================================================

File: cpp/base/src/main/vsdk/toolkit/media/ShapeDescriptor.cpp
#include "ShapeDescriptor.h"
#include "java/lang/String.h"

ShapeDescriptor::ShapeDescriptor(const java::String* label_) {
    if (label_ != nullptr) {
        label = new java::String(*label_);
    } else {
        label = nullptr;
    }
}

ShapeDescriptor::~ShapeDescriptor() {
    if (label != nullptr) {
        delete label;
        label = nullptr;
    }
}

java::String* ShapeDescriptor::getLabel() const {
    if (label != nullptr) {
        return new java::String(*label);
    }
    return nullptr;
}

void ShapeDescriptor::setLabel(const java::String* label_) {
    if (label != nullptr) {
        delete label;
    }
    if (label_ != nullptr) {
        label = new java::String(*label_);
    } else {
        label = nullptr;
    }
}
